src/vcpkg/vcpkgpaths.cpp

```diff
@@ -672,9 +679,39 @@ namespace vcpkg
                                                        m_pimpl->m_config_dir,
                                                        *this);
 
+            auto make_relative_to_manifest = [&](std::string overlay_path) -> std::string {
+                auto const p = Path(overlay_path);
+                if (p.is_relative())
+                {
+                    Path ret = m_pimpl->m_config.directory / overlay_path;
+                    return ret.c_str();
+                }
+                return overlay_path;
+            };
+
+            if (!m_pimpl->m_config.directory.empty())
+            {
+                auto& config_ports = m_pimpl->m_config.config.overlay_ports;
+                auto& config_triplets = m_pimpl->m_config.config.overlay_triplets;
+                Util::transform(config_ports, make_relative_to_manifest);
+                Util::transform(config_triplets, make_relative_to_manifest);
```

Comment:
I don't think the extra locals are adding much here but this is an extremely mild preference of mine. But there is a local that could help:

```suggestion
                auto& config = m_pimpl->m_config.config;
                Util::transform(config.overlay_ports, resolve_relative_to_config);
                Util::transform(config.overlay_triplets, resolve_relative_to_config);
```

Alternately, does it make sense to keep those and reuse them below as:

```
            overlay_ports = merge_overlays(
                args.cli_overlay_ports, config_ports, args.env_overlay_ports);
            overlay_triplets = merge_overlays(
                args.cli_overlay_triplets, config_triplets, args.env_overlay_triplets);
```